toqito/channel_metrics/completely_bounded_trace_norm.py

```diff
@@ -0,0 +1,66 @@
+"""Compute the completely bounded trace norm of a quantum channel."""
+import numpy as np
+import cvxpy as cp
+
+from toqito.channel_ops import apply_channel, dual_channel
+from toqito.channel_props import is_quantum_channel, is_completely_positive
+from toqito.state_metrics import trace_norm
+
+def completely_bounded_trace_norm(phi: np.ndarray) -> float:
+    r"""
+    Compute the completely bounded trace norm / diamond norm of a quantum channel [WatCBNorm18].
+    The algorithm in p.11 of [WatSDP12] with implementation in QETLAB [JohQET] is used.
+
+    References
+    ==========
+    .. [WatCNorm18] : Watrous, John.
+    “The theory of quantum information.” Section 3.3.2: “The completely bounded trace norm”.
+    Cambridge University Press, 2018.
+
+    .. [WatSDP09]:   Watrous, John.
+    "Simpler semidefinite programs for completely bounded norms"
+    https://arxiv.org/pdf/1207.5726.pdf
+
+    .. [JohQET]: Nathaniel Johnston. QETLAB:
+    A MATLAB toolbox for quantum entanglement, version 0.9.
+    https://github.com/nathanieljohnston/QETLAB/blob/master/DiamondNorm.m
+    http://www.qetlab.com, January 12, 2016. doi:10.5281/zenodo.44637
+
+    :raises ValueError: If matrix is not square.
+    :param phi: superoperator as choi matrix
+    :return: The completely bounded trace norm of the channel
+    """
+    dim_Lx, dim_Ly = phi.shape
+
+    if dim_Lx != dim_Ly:
```

Comment:
The way you are doing this here is completely fine, although `toqito` does have a `matrix_props.is_square` function that can also be invoked. Your call as to whether or not you'd like to use that here or not.